### PR TITLE
[TB-3062] Add reader mode settings error handling styles and colors

### PR DESCRIPTION
**Pull request by MichaelKMalak**
### What 🕵️ 🔍

- Add reader mode settings error-handling styles and colors
- Implement `AppMenu`

### References 📝 🔗

- [TB-3062](https://xainag.atlassian.net/browse/TB-3062)

----------

### CODE REVIEW

File: lib/src/linden/xcolors.dart [hunk 1]
```diff
@@ -40,6 +40,8 @@ class XColors {
   static const _orange = Color(0xffF29100);
   static const _white = Color(0xffffffff);
   static const _beige = Color(0xfff8f0e3);
+  static const _pink20 = Color(0xffF3CDDE);
+  static const _pinkNew = Color(0xffDB1870);
```
Comment (peterstojanowski) [inline lib/src/linden/xcolors.dart:44]: We should probably use names from the Design System.

File: lib/src/widget/menu/app_menu.dart [hunk 2]
```diff
@@ -0,0 +1,152 @@
+import 'package:flutter/material.dart';
+import 'package:xayn_design/src/linden/linden.dart';
+import 'package:xayn_design/src/widget/unter_den_linden/unter_den_linden.dart';
+
+const _kDividerThickness = 1.0;
+
+class AppMenu extends StatelessWidget {
```
Comment (peterstojanowski) [inline lib/src/widget/menu/app_menu.dart:7]: Would be good to add a comment what is `AppMenu` and what it does.

File: lib/src/widget/menu/app_menu.dart [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+          'AppMenu needs to have horizontal constraints: start or end or width',
+        ),
+        assert(
+          top == null || bottom == null || height == null,
+          'AppMenu needs to have vertical constraints: top or bottom or height',
+        ),
+        super(key: key);
+
+  final List<Widget> children;
+  final String? errorStr;
```
Comment (peterstojanowski) [inline lib/src/widget/menu/app_menu.dart:34]: What is `errorStr` used for?

File: lib/src/linden/xcolors.dart (hunk 1 above)
Comment (MichaelKMalak) [reply lib/src/linden/xcolors.dart:44]: yeah, it's called `_pink` in the new design system, and we already have a `_pink` here, I named it `_pinkNew` 🤷‍♂️

File: lib/src/widget/menu/app_menu.dart (hunk 3 above)
Comment (MichaelKMalak) [reply lib/src/widget/menu/app_menu.dart:34]: menus [display error in a specific way](https://user-images.githubusercontent.com/12827629/154662894-18a112e3-31d0-43a5-88a2-de504e2e14a0.png) according to the design team.
I'll add a comment there.

File: lib/src/linden/xcolors.dart (hunk 1 above)
Comment (peterstojanowski) [reply lib/src/linden/xcolors.dart:44]: Then update the old color to use the new hex value. I will be working on updating the colors soon anyway.

Comment (MichaelKMalak) [reply lib/src/linden/xcolors.dart:44]: didn't know it's ok to do so.. will do now.

File: lib/src/widget/menu/app_menu.dart (hunk 3 above)
Comment (peterstojanowski) [reply lib/src/widget/menu/app_menu.dart:34]: Maybe rename it to `errorMessage`? `errorStr` sounds a bit cryptic.

Comment (MichaelKMalak) [reply lib/src/widget/menu/app_menu.dart:34]: alright 👍

File: lib/src/widget/menu/app_menu.dart (hunk 2 above)
Comment (MichaelKMalak) [reply lib/src/widget/menu/app_menu.dart:10]: [a060c10](https://github.com/xaynetwork/xayn_design/pull/69/commits/a060c10fbff75619a61ccb7ac0e144e1461acc70)

File: lib/src/linden/xcolors.dart (hunk 1 above)
Comment (MichaelKMalak) [reply lib/src/linden/xcolors.dart]: [a060c10](https://github.com/xaynetwork/xayn_design/pull/69/commits/a060c10fbff75619a61ccb7ac0e144e1461acc70)

**Review by peterstojanowski — APPROVED** [review]
Looks good 👍

**Pull request merged**